src/lib/disk_cache/lmdb/disk_cache.ml

```diff
@@ -39,9 +39,14 @@ module Make (Data : Binable.S) = struct
     Rw.get ~env db idx |> Option.value_exn
 
   let put ({ env; db; counter; garbage } : t) (x : Data.t) : id =
-    (* TODO: we may reuse IDs by pulling them from the `garbage` hash set *)
-    let idx = !counter in
-    incr counter ;
+    let idx =
+      match Hash_set.find garbage ~f:(const true) with
```

Comment:
I don't understand what you are trying to achieve by using `~f:(const true)`.
This will always return either a random element (or the first, I don't remember if `find` on `Hashtbl` is deterministic) if the set is empty (which is the case until the GC has been executed at least once on a value) or it will increase the counter

Reply:
selecting a random element from hashset and use it as new key is exactly the expected behavior. Any key in that set is abandoned